### PR TITLE
Add booking screen

### PR DESCRIPTION
**Pull request by lelouchlamperouge18**

### CODE REVIEW

File: src/screens/Appointments/AppointmentBooking.tsx [hunk 1]
```diff
@@ -1,25 +1,110 @@
...
-} from 'react-native';
+  appointmentOptionsDate,
+  appointmentOptionsTime,
+} from '../../constants/Appointment.constant';
+import ButtonMain from '../../components/button/ButtonMain';
+import AppointmentOption from '../../components/appointmentItem/AppointmentOption';
 
-const AppointmentBooking = (): JSX.Element => {
+const AppointmentBooking = ({navigation}: NavigationOptions): JSX.Element => {
+  const [open, setOpen] = useState(false);
```
Comment (LofizDev) [inline src/screens/Appointments/AppointmentBooking.tsx]: - open => isShowListTopic 
- missing type

File: src/screens/Appointments/AppointmentBooking.tsx [hunk 2]
```diff
@@ -1,25 +1,110 @@
...
+  const [value, setValue] = useState(topics[1].topicName);
```
Comment (LofizDev) [inline src/screens/Appointments/AppointmentBooking.tsx]: - value => selectedTopic
- missing type

File: src/screens/Appointments/AppointmentBooking.tsx [hunk 3]
```diff
@@ -1,25 +1,110 @@
...
+  const items = topics.map(topic => ({
+    label: topic.topicName,
+    value: topic.topicName,
+  }));
+  const [openDoctor, setOpenDoctor] = useState(false);
```
Comment (LofizDev) [inline src/screens/Appointments/AppointmentBooking.tsx]: - openDoctor => isShowListDoctor
- missing type

File: src/screens/Appointments/AppointmentBooking.tsx [hunk 4]
```diff
@@ -1,25 +1,110 @@
...
+  const [valueDoctor, setValueDoctor] = useState(doctor[1].doctorName);
```
Comment (LofizDev) [inline src/screens/Appointments/AppointmentBooking.tsx]: valueDoctor => selectedDoctor

File: src/screens/Appointments/AppointmentBooking.tsx [hunk 5]
```diff
@@ -1,25 +1,110 @@
...
+  appointmentOptionsTime,
+} from '../../constants/Appointment.constant';
+import ButtonMain from '../../components/button/ButtonMain';
+import AppointmentOption from '../../components/appointmentItem/AppointmentOption';
 
-const AppointmentBooking = (): JSX.Element => {
+const AppointmentBooking = ({navigation}: NavigationOptions): JSX.Element => {
+  const [open, setOpen] = useState(false);
+  const [value, setValue] = useState(topics[1].topicName);
+  const items = topics.map(topic => ({
```
Comment (LofizDev) [inline src/screens/Appointments/AppointmentBooking.tsx]: - rename item to make sense
 - missing type at topic

File: src/screens/Appointments/AppointmentBooking.tsx [hunk 6]
```diff
@@ -1,25 +1,110 @@
...
+    label: topic.topicName,
+    value: topic.topicName,
+  }));
+  const [openDoctor, setOpenDoctor] = useState(false);
+  const [valueDoctor, setValueDoctor] = useState(doctor[1].doctorName);
+  const itemsDoctor = doctor.map(item => ({
```
Comment (LofizDev) [inline src/screens/Appointments/AppointmentBooking.tsx]: - rename itemDoctor to make sense
- why doctor.map instead doctors ?
- rename item & add type

File: src/components/appointmentItem/AppointmentOption.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import {StyleSheet, View, Text} from 'react-native';
+
+import {spacing, theme} from '../../theme/index';
+
+type AppointmentOptionProps = {
+  text?: string;
+  day?: string;
+};
```
Comment (LofizDev) [inline src/components/appointmentItem/AppointmentOption.tsx]: could you please tell me why optional here
move this one AppointmentOptionProps to model and import

File: src/model/Appointment.model.ts [hunk 8]
```diff
@@ -11,3 +11,12 @@ export type AppointmentCardItem = Omit<
   AppointmentsCalendarCard,
   'status' | 'doctorName'
 >;
+
+export type AppointmentOptionsDate = {
+  text: string;
```
Comment (LofizDev) [inline src/model/Appointment.model.ts:16]: could you please rename text to make sense

File: src/screens/Appointments/AppointmentBooking.tsx [hunk 9]
```diff
@@ -1,25 +1,110 @@
...
+          />
+        </View>
+        <Text style={styles.fieldTitle}>Appointment type</Text>
+        <View style={styles.optionAppointment}>
+          <AppointmentOption text="Chat" />
+          <AppointmentOption text="Video" />
+        </View>
+        <Text style={styles.fieldTitle}>Select Date</Text>
+        <View style={styles.optionAppointment}>
+          {appointmentOptionsDate.map((item, index) => (
```
Comment (LofizDev) [inline src/screens/Appointments/AppointmentBooking.tsx]: - item => appointment
 - missing type 
 - missing key

File: src/screens/Appointments/AppointmentBooking.tsx [hunk 10]
```diff
@@ -1,25 +1,110 @@
...
+            <AppointmentOption text={item.text} day={item.day} />
+          ))}
+        </View>
+        <Text style={styles.fieldTitle}>Select Time</Text>
+        <View style={styles.optionAppointment}>
+          {appointmentOptionsTime.map((item, index) => (
```
Comment (LofizDev) [inline src/screens/Appointments/AppointmentBooking.tsx]: item => appointment
missing type
missing key

File: src/components/appointmentItem/AppointmentOption.tsx (hunk 7 above)
Comment (sontego810) [reply src/components/appointmentItem/AppointmentOption.tsx]: optional because we have 3 types of option (look at design)

**Pull request merged**